$NetBSD: patch-CMakeLists.txt,v 1.3 2025/01/22 12:47:49 wiz Exp $

First part: use proper variable for boost include directories

Rest:
https://github.com/dscharrer/innoextract/commit/045f72a73d692bb3a918650e33106df5233f90c9

--- CMakeLists.txt.orig	2020-08-09 18:48:04.000000000 +0000
+++ CMakeLists.txt
@@ -160,11 +160,15 @@ find_package(Boost REQUIRED COMPONENTS
 )
 list(APPEND LIBRARIES ${Boost_LIBRARIES})
 link_directories(${Boost_LIBRARY_DIRS})
-include_directories(SYSTEM ${Boost_INCLUDE_DIR})
+include_directories(SYSTEM ${Boost_INCLUDE_DIRS})
 if(NOT Boost_VERSION_MACRO)
 	# CMP0093 changed Boost_VERSION to x.y.z format and provide the old format in Boost_VERSION_MACRO
 	set(Boost_VERSION_MACRO ${Boost_VERSION})
 endif()
+add_definitions(-DBOOST_SYSTEM_DISABLE_THREADS)
+if(win32)
+	add_definitions(-DBOOST_SYSTEM_USE_UTF8)
+endif()
 
 has_static_libs(Boost Boost_LIBRARIES)
 if(Boost_HAS_STATIC_LIBS)
